mesonbuild/compilers/mixins/gnu.py

```diff
@@ -629,6 +629,22 @@ def use_linker_args(cls, linker: str, version: str) -> T.List[str]:
             return ['-fuse-ld=mold']
         return super().use_linker_args(linker, version)
 
+    def get_lto_link_args(self, *, threads: int = 0, mode: str = 'default',
+                          thinlto_cache_dir: T.Optional[str] = None) -> T.List[str]:
+        args: T.List[str] = []
+        if threads == 0:
+            if self._has_lto_auto_support:
+                args.append('-flto=auto')
+            else:
+                # This matches gcc's behavior of using the number of cpus, but
+                # obeying meson's MESON_NUM_PROCESSES convention.
+                args.append(f'-flto={mesonlib.determine_worker_count()}')
+        elif threads > 0:
+            args.append(f'-flto={threads}')
+        else:
+            args.extend(super().get_lto_compile_args(threads=threads))
```

Comment:
Probably a moot point. We have super().get_lto_compile_args, which for gnulikecompiler is... hardcoded to `-flto`. Meanwhile, get_lto_link_args is, as per the reason for this PR, stuck at... `-flto`. Which we get from linkers/linkers.py. The abstractions here feel weird.

Reply:
So just `-flto`? Yeah, that's fine.